Source/Engine/PackageBuilder/PatternLinker.cs

```diff
@@ -50,33 +46,14 @@ protected internal override Syntax VisitPackage(PackageSyntax node)
             ReadOnlyCollection<RequiredPackageSyntax> requiredPackages = Visit(node.RequiredPackages);
             CheckDuplicatedPatternNames(node);
             ReadOnlyCollection<Syntax> rootPatterns = Visit(node.Patterns);
-            List<Syntax> searchTargetsFromPatterns = null;
-            foreach (PatternSyntax p in fAllPatterns.Where(x => ((PatternSyntax)x).IsSearchTarget))
-            {
-                var t = Syntax.SearchTarget(p.FullName);
-                if (searchTargetsFromPatterns == null)
-                    searchTargetsFromPatterns = new List<Syntax>();
-                searchTargetsFromPatterns.Add(t);
-            }
-            ReadOnlyCollection<Syntax> searchTargets;
-            if (searchTargetsFromPatterns != null)
-            {
-                var newSearchTargets = new List<Syntax>(node.SearchTargets);
-                newSearchTargets.AddRange(searchTargetsFromPatterns);
-                searchTargets = new ReadOnlyCollection<Syntax>(newSearchTargets);
-            }
-            else
-                searchTargets = node.SearchTargets;
-            searchTargets = Visit(searchTargets);
+            ReadOnlyCollection<Syntax> searchTargets = Visit(node.SearchTargets);
             foreach (PatternReferenceInContext x in fPatternReferences)
                 ResolvePatternReference(x);
-            if (fExtractedPatterns.Count > 0)
-                rootPatterns = new ReadOnlyCollection<Syntax>(rootPatterns.Union(fExtractedPatterns).ToArray());
             LinkedPackageSyntax result = Syntax.LinkedPackage(requiredPackages, searchTargets, rootPatterns);
+            result.TextRange = node.TextRange;
             fAllPatterns.Clear();
```

Comment:
I think `fAllPatterns` is not needed here anymore. It is only updated, but never read.